### PR TITLE
Exceptions

### PR DESCRIPTION
**Pull request by OleksandrZheleznyi**

### CODE REVIEW

File: Exceptions/Program.cs [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+            Worker[] workers = new Worker[count];
+
+            workers[0] = new Worker("Antonov", "AA", "trainee", 2014);
+            workers[1] = new Worker("Vadimov", "DD", "junior", 2003);
+            workers[2] = new Worker("Ivanchenko", "TT", "middle", 1998);
+            workers[3] = new Worker("Abrasyan", "AB", "senior", 1995);
+            workers[4] = new Worker("Zuravko", "VD", "essential", 2001);
+
+            //Sort struct by surname
+            Array.Sort(workers, new Comparison<Worker>((c, d) => c.surname.CompareTo(d.surname)));
```
Comment (OleksandrIvanovTheBest) [inline Exceptions/Program.cs]: It's not your code. It's a copypaste. Why do you need new Comparison?

File: Exceptions/Program.cs [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+            Price[] products =
+            {
+                new Price("Shauma", "Silpo", 56.8),
+                new Price("Ariel", "EKO", 83.7),
+                new Price("Gala", "ATB", 25.8),
+                new Price("Tide", "Novus", 50)
+            };
+
+            //sort struct product by magazine
+            Array.Sort(products, new Comparison<Price>((c, d) => c.magazine.CompareTo(d.magazine)));
```
Comment (OleksandrIvanovTheBest) [inline Exceptions/Program.cs]: same

File: Exceptions/Program.cs [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+
+            foreach (Price product in products)
+            {
+                product.Show();
+            }
+
+            string productName = "Mulo";
+
+            Exception ex = new Exception("Product does not exist");
```
Comment (OleksandrIvanovTheBest) [inline Exceptions/Program.cs]: Don't need variable.

**Review by OleksandrIvanovTheBest — CHANGES_REQUESTED** [review]

File: Exceptions/Program.cs (hunk 1 above)
Comment (OleksandrZheleznyi) [reply Exceptions/Program.cs]: Yes, its copypaste, I wanted sort workers by field surname, but it does not affect the solution of the task. I remove this code.

Comment (OleksandrIvanovTheBest) [reply Exceptions/Program.cs]: Why not just Array.Sort(workers, (Worker c, Worker d) => c.surname.CompareTo(d.surname)); ?

File: Exceptions/Program.cs [hunk 4]
```diff
@@ -81,7 +61,7 @@ static void Main(string[] args)
                     }
                 }
                 if (!hasFoundValue)
-                    throw ex;
+                    throw new Exception("Product does not exist"); ;
```
Comment (OleksandrIvanovTheBest) [inline Exceptions/Program.cs]: Create custom Exception

File: Exceptions/Program.cs [hunk 5]
```diff
@@ -57,18 +47,8 @@ static void Main(string[] args)
                 new Price("Tide", "Novus", 50)
             };
 
-            //sort struct product by magazine
-            Array.Sort(products, new Comparison<Price>((c, d) => c.magazine.CompareTo(d.magazine)));
```
Comment (OleksandrIvanovTheBest) [inline Exceptions/Program.cs]: Resurrect this code. Just change the sorting code

File: Exceptions/Program.cs [hunk 6]
```diff
@@ -17,16 +17,6 @@ static void Main(string[] args)
             workers[3] = new Worker("Abrasyan", "AB", "senior", 1995);
             workers[4] = new Worker("Zuravko", "VD", "essential", 2001);
 
-            //Sort struct by surname
-            Array.Sort(workers, new Comparison<Worker>((c, d) => c.surname.CompareTo(d.surname)));
```
Comment (OleksandrIvanovTheBest) [inline Exceptions/Program.cs]: Same

**Review by OleksandrIvanovTheBest — CHANGES_REQUESTED** [review]
Don't remove sorting

File: Exceptions/Program.cs (hunk 6 above)
Comment (OleksandrIvanovTheBest) [reply Exceptions/Program.cs]: Use Linq

**Review by OleksandrIvanovTheBest — DISMISSED** [review]

**Review by OleksandrIvanovTheBest — APPROVED** [review]

**Pull request merged**